OttoTest/app/src/main/java/com/example/raulstriglio/ottotest/OttoFragment.java

```diff
@@ -11,56 +11,92 @@
 
 import com.example.raulstriglio.ottotest.API.MyApi;
 import com.example.raulstriglio.ottotest.Events.CustomUserEvent;
+import com.example.raulstriglio.ottotest.Utilities.MyBus;
 import com.example.raulstriglio.ottotest.model.User;
-import com.squareup.otto.Bus;
 import com.squareup.otto.Subscribe;
 
+import java.util.ArrayList;
+
+import javax.inject.Inject;
+
 /**
  * Created by raul.striglio on 28/07/16.
  */
 public class OttoFragment extends Fragment {
 
-    private String URL = "https://raw.githubusercontent.com/restriglio/OttoTest/master/json/info.json";
-    View fragmentView;
-    Button getData;
-    MyApi myApi;
-    Bus bus;
-    TextView id_name, id_lastname;
+    private final String URL = "https://raw.githubusercontent.com/restriglio/OttoTest/master/json/info.json";
+    private View fragmentView;
+    private Button getData;
+
+    @Inject
+    protected MyApi myApi;
+    private TextView id_name, id_lastname;
 
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         fragmentView = inflater.inflate(R.layout.otto_fragment, container, false);
-
         getData = (Button) fragmentView.findViewById(R.id.getData);
         id_name = (TextView) fragmentView.findViewById(R.id.id_name);
         id_lastname = (TextView) fragmentView.findViewById(R.id.id_lastname);
-
-        myApi = new MyApi(getContext());
-
         getData.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                //get data
-                myApi.sedRequest(URL);
+                myApi.readFromFile();
             }
         });
-
         return fragmentView;
     }
 
     @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((OttoTestApplication) getActivity().getApplication()).getMyApiComponent().inject(this);
+    }
+
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        if (!(getActivity() instanceof ICallback)) {
+            throw new ClassCastException(getActivity().toString()
+                    + getResources().getString(R.string.OttoFragmentError));
```

Comment:
Is not necessary to use a resource here since this error will only be shown in the console. Always use resources for UI texts.